File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Skills/SkillRouter.cs
namespace Microsoft.Bot.Solutions.Skills
{
    using System.Collections.Generic;
    using System.Linq;

    public class SkillRouter
    {
        private List<SkillDefinition> _registeredSkills;

        public SkillRouter(List<SkillDefinition> registeredSkills)
        {
            // Retrieve any Skills that have been registered with the Bot
            _registeredSkills = registeredSkills;
        }

        public SkillDefinition IdentifyRegisteredSkill(string skillName)
        {
            SkillDefinition matchedSkill = null;

            // Did we find any skills?
            if (_registeredSkills != null)
            {
                // Identify a skill by taking the LUIS model name identified by the dispatcher and matching to the skill luis model name
                // Bug raised on dispatcher to move towards LuisModelId instead perhaps?
                matchedSkill = _registeredSkills.FirstOrDefault(s => s.DispatchIntent == skillName);
                return matchedSkill;
            }
            else
            {
                return null;
            }
        }
    }
}

File: lib/typescript/botbuilder-solutions/src/skills/skillRouter.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { SkillDefinition } from './skillDefinition';

export class SkillRouter {

    private registeredSkills: SkillDefinition[];

    constructor(registeredSkills: SkillDefinition[]) {
        // Retrieve any Skills that have been registered with the Bot
        this.registeredSkills = registeredSkills;
    }
    public identifyRegisteredSkill(skillName: string): SkillDefinition | undefined {

        let matchedSkill: SkillDefinition;

        // Did we find any skills?
        if (this.registeredSkills !== undefined) {
            // Identify a skill by taking the LUIS model name identified by the dispatcher and matching to the skill luis model name
            // Bug raised on dispatcher to move towards LuisModelId instead perhaps?
            matchedSkill = <SkillDefinition>this.registeredSkills.find((s: SkillDefinition) => s.dispatchIntent === skillName);

            return matchedSkill;
        } else {

            return undefined;
        }
    }
}
